GPflow/kernels.py

```diff
@@ -51,21 +80,45 @@ def __init__(self, input_dim, active_dims=None):
         self.input_dim = input_dim
         if active_dims is None:
             self.active_dims = slice(input_dim)
+        elif type(active_dims) is slice:
+            self.active_dims = active_dims
+            if active_dims.start is not None and active_dims.stop is not None and active_dims.step is not None:
+                assert len(range(*active_dims)) == input_dim  # pragma: no cover
         else:
             self.active_dims = np.array(active_dims, dtype=np.int32)
             assert len(active_dims) == input_dim
 
+        self.num_gauss_hermite_points = 20
+
     def _slice(self, X, X2):
         if isinstance(self.active_dims, slice):
             X = X[:, self.active_dims]
             if X2 is not None:
                 X2 = X2[:, self.active_dims]
-            return X, X2
         else:
             X = tf.transpose(tf.gather(tf.transpose(X), self.active_dims))
             if X2 is not None:
                 X2 = tf.transpose(tf.gather(tf.transpose(X2), self.active_dims))
-            return X, X2
+        with tf.control_dependencies([
+            tf.assert_equal(tf.shape(X)[1], tf.constant(self.input_dim, dtype=settings.dtypes.int_type))
+        ]):
+            X = tf.identity(X)
+
+        return X, X2
+
+    def _slice_cov(self, cov):
```

Comment:
Would be great to have docstrings for both slice and slice_cov (though I realize _slice is not part of this PR.)